### PR TITLE
lib/options: `defaultNullOpts` no longer requires a "default" or "description"

### PR DESCRIPTION
**Pull request by MattSturgeon**
- **lib/options: `defaultNullOpts` don't require having a default**

Made the `default` and `description` arguments optional for all **prime variants** of `defaultNullOpts` functions.

If neither `default` nor `description` are provided, the option will have no description.

The "Plugin default" line is only added when a `default` argument is present (`args ? default`).

This means we can now use `defaultNullOpts`, even when we explicitly _do not_ want to document a "plugin default".

- **docs: emphasise "Plugin default" to match nixpkgs**

Match the style of similar headings that come from nixpkgs. This tiny commit could be **backported** to 24.05.

<details><summary><strong>Docs render</strong></summary>
<p>

image

</p>
</details>

### CODE REVIEW

**Review by GaetanLepage — APPROVED** [review]
Goof for me.
Let's wait for an approval from @traxys.

**Review by traxys — APPROVED** [review]

**Pull request merged**
